### PR TITLE
[Bug-12410] [API]Fix the worker list result in workflow definition only has default

### LINKED ISSUE #12377: [Bug] [api-service] Tenant user run Workflow  WorkerGroups  can't find admin add  other Worker Group list
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

version :3.1.0

 admin  add new  Workflow  

image

tenant user  hadoop   run  Workflow  

image

### What you expected to happen

 i can Read the The source code  find    use this  api 
`dolphinscheduler/worker-groups/all`

WorkerGroupServiceImpl.queryAllGroup
image

ResourceAcquisitionAndPermissionCheck.userOwnedResourceIdsAcquisition

image

if use  tenant user   only  return ` Collections.emptySet();` why  ???  i can  find  subsequent  worker-groups add ZOOKEEPER  `/dolphinscheduler/nodes/worker/`

image

### How to reproduce

 I used to use version 2.0   Tenant user  can find  admin add new WorkerGroups  

How can I discover newly added  WorkerGroups  

### Anything else

_No response_

### Version

3.1.x

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### LINKED ISSUE #12410: [Bug] [API] In the workflow definition, the result of the worker list is only default
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

In the workflow definition, the result of the worker list is only default

### What you expected to happen

In the workflow definition, the result of the worker list is only default

### How to reproduce

In the workflow definition, the list of workers appears fine

### Anything else

_No response_

### Version

3.1.x

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zhuangchong**
## Purpose of the pull request

this pr close: #12410
this pr close: #12377

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/permission/ResourcePermissionCheckServiceImpl.java [hunk 1]
```diff
@@ -371,13 +370,17 @@ public List<AuthorizationType> authorizationTypes() {
 
         @Override
         public boolean permissionCheck(int userId, String url, Logger logger) {
-            return false;
+            return true;
```
Comment (caishunfeng) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/permission/ResourcePermissionCheckServiceImpl.java]: PTAL @WangJPLeo

Comment (ruanwenjun) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/permission/ResourcePermissionCheckServiceImpl.java]: I find we return false in other methods in `ResourcePermissionCheckServiceImpl`, we may need to check this.
e.g. `AlertPluginInstanceResourceList`, `AlertGroupResourceList`, `TenantResourceList`.

Comment (WangJPLeo) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/permission/ResourcePermissionCheckServiceImpl.java]: This is the permission control to control the WorkerGroup DML operation. Currently, other users except for Admin have no permission by default, so this should not be changed to true.

**Comment by caishunfeng** [comment]
Hi @zhuangchong if this issue exist in 3.0.x too, please add the label `3.0.x`.

**Comment by zhuangchong** [comment]
> Hi @zhuangchong if this issue exist in 3.0.x too, please add the label `3.0.x`.

3.0.x version does not add this permissions detection.

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/permission/ResourcePermissionCheckServiceImpl.java [hunk 2]
```diff
@@ -467,9 +462,6 @@ public boolean permissionCheck(int userId, String url, Logger logger) {
 
         @Override
         public Set<Integer> listAuthorizedResource(int userId, Logger logger) {
-            if (userId != 0) {
```
Comment (WangJPLeo) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/permission/ResourcePermissionCheckServiceImpl.java:471]: Release means all users can access, In fact, all non admin users have no access and operation rights to the data resources in the security center before. @zhuangchong PTAL

Comment (zhuangchong) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/permission/ResourcePermissionCheckServiceImpl.java:471]: This is tenant data. When saving the workflow definition, you need to select a tenant (users of historical versions can query all tenant data), so at present, ordinary users should not be restricted to query tenant permissions.

Comment (ruanwenjun) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/permission/ResourcePermissionCheckServiceImpl.java:471]: I find we didn't has tenant permission in DS, so we need to remove this judge here.

Comment (zhuangchong) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/permission/ResourcePermissionCheckServiceImpl.java:471]: Yes, it should be compatible with history, user can query all tenants.

**Review by ruanwenjun — DISMISSED** [review]
LGTM

**Review by ruanwenjun — APPROVED** [review]

**Review by caishunfeng — APPROVED** [review]
+1

**Pull request merged**